test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs

```diff
@@ -105,22 +133,30 @@ public void TagHelperDescrictorsAreCompatibleWithPinnedVisualStudioVersion()
             Assert.Equal(expectedDTDescriptor.Remarks, dtDescriptor.Remarks, StringComparer.Ordinal);
             Assert.Equal(expectedDTDescriptor.Summary, dtDescriptor.Summary, StringComparer.Ordinal);
 
-            var attribute = Assert.Single(vsDescriptor.Attributes);
-            var expectedAttribute = Assert.Single(expectedVSDescriptor.Attributes);
-            Assert.Equal(attribute.IsIndexer, expectedAttribute.IsIndexer);
-            Assert.Equal(attribute.IsStringProperty, expectedAttribute.IsStringProperty);
-            Assert.Equal(attribute.Name, expectedAttribute.Name, StringComparer.Ordinal);
-            Assert.Equal(attribute.PropertyName, expectedAttribute.PropertyName, StringComparer.Ordinal);
-            Assert.Equal(attribute.TypeName, expectedAttribute.TypeName, StringComparer.Ordinal);
+            var attributes = vsDescriptor.Attributes.ToArray();
+            var expectedAttributes = expectedVSDescriptor.Attributes.ToArray();
+            Assert.Equal(attributes.Length, expectedAttributes.Length);
 
-            var dtAttribute = attribute.DesignTimeDescriptor;
-            var expectedDTAttribute = expectedAttribute.DesignTimeDescriptor;
-            Assert.Equal(dtAttribute.Remarks, expectedDTAttribute.Remarks, StringComparer.Ordinal);
-            Assert.Equal(dtAttribute.Summary, expectedDTAttribute.Summary, StringComparer.Ordinal);
+            for (var i = 0; i < attributes.Length; i++)
+            {
+                var attribute = attributes[i];
+                var expectedAttribute = expectedAttributes[i];
+                Assert.Equal(attribute.IsIndexer, expectedAttribute.IsIndexer);
+                Assert.Equal(attribute.IsEnum, expectedAttribute.IsEnum);
+                Assert.Equal(attribute.IsStringProperty, expectedAttribute.IsStringProperty);
+                Assert.Equal(attribute.Name, expectedAttribute.Name, StringComparer.Ordinal);
+                Assert.Equal(attribute.PropertyName, expectedAttribute.PropertyName, StringComparer.Ordinal);
+                Assert.Equal(attribute.TypeName, expectedAttribute.TypeName, StringComparer.Ordinal);
+
+                var dtAttribute = attribute.DesignTimeDescriptor;
+                var expectedDTAttribute = expectedAttribute.DesignTimeDescriptor;
+                Assert.Equal(dtAttribute.Remarks, expectedDTAttribute.Remarks, StringComparer.Ordinal);
+                Assert.Equal(dtAttribute.Summary, expectedDTAttribute.Summary, StringComparer.Ordinal);
```

Comment:
Can't this be done more simply with a comparator and the `IEnumerable` assertions?